src/Package/Impl/ProjectSystem/RDebugLaunchProvider.cs

```diff
@@ -85,31 +87,126 @@ public RDebugLaunchProvider(ConfiguredProject configuredProject, IRInteractiveWo
+            } else {
+                fExists = FileSystem.FileExists(file);
+            }
+
+            if (!fExists) {
                 _interactiveWorkflow.ActiveWindow?.InteractiveWindow.WriteErrorLine(errorMessage);
                 return;
             }
+
+            await Callbacks.WriteConsoleEx($"Sourcing: {file}\n", OutputType.Output, CancellationToken.None);
             await _interactiveWorkflow.Operations.SourceFileAsync(file, echo: false).SilenceException<Exception>();
         }
+
+        private async Task SendProjectAsync(EnvDTE.Project project, string remotePath, string filterString) {
+            await Callbacks.WriteConsoleEx("Preparing to transfer project.\n", OutputType.Output, CancellationToken.None);
+
+            var projectDir = Path.GetDirectoryName(project.FullName);
+            var projectName = Path.GetFileNameWithoutExtension(project.FullName);
+            var filter = new TransferFileFilter(filterString);
+
+            await Callbacks.WriteConsoleEx($"Remote destination: {remotePath}\n", OutputType.Output, CancellationToken.None);
+            await Callbacks.WriteConsoleEx($"File filter applied: {filterString}\n", OutputType.Output, CancellationToken.None);
+            await Callbacks.WriteConsoleEx("Compressing project files for transfer:\n", OutputType.Output, CancellationToken.None);
+
+            var compressedFilePath = await Task.Run(() => FileSystem.CompressDirectory(projectDir, (p) => {
+                Callbacks.WriteConsoleEx($"Compressing: {p}\n", OutputType.Output, CancellationToken.None).Wait();
+                return filter.Match(p);
+            }));
+
+            using (var fts = new FileTransferSession(Session, FileSystem)) {
+                await Callbacks.WriteConsoleEx("Transferring project to remote host...", OutputType.Output, CancellationToken.None);
+                var remoteFile = await fts.SendFileAsync(compressedFilePath);
+                await Session.EvaluateAsync<string>(Invariant($"rtvs:::save_project({remoteFile.Id}, {projectName.ToRStringLiteral()}, '{remotePath.ToRPath()}')"), REvaluationKind.Normal);
+            }
+
+            await Callbacks.WriteConsoleEx(" Completed.\n", OutputType.Output, CancellationToken.None);
+        }
+
+        private async Task<string> GetStartupFileAsync(bool transferFiles, EnvDTE.Project project) {
+            if (transferFiles && Session.IsRemoteSession) { // remote
+                var projectName = Path.GetFileNameWithoutExtension(project.FullName);
+                var remotePath = (await _properties.GetRemoteProjectPathAsync()).ToRPath();
+                var startUpFile = (await _properties.GetStartupFileAsync()).ToRPath();
+                return remotePath + projectName + "/" + startUpFile;
+            } else { // local
+                var projDir = Path.GetDirectoryName(project.FullName);
```

Comment:
Do we need to do anything if `transferFiles` is `true`, but `Session.IsRemoteSession` is false